web/ui/src/common/resourcesFactory.js

```diff
@@ -269,16 +287,56 @@
 
             return function(/* args */){
                 var url = config.url.apply(null, arguments),
-                    payload;
+                    resourceName = url,
+                    payload,
+                    // deferred that will be returned to the user
+                    deferred = $q.defer();
+
+                // if resourceName has query params, strip em off
+                if(resourceName.indexOf("?")){
+                    resourceName = resourceName.split("?")[0];
+                }
+
+                // NOTE: all of our code expects angular's wrapped
+                // promises which provide a success and error method
+                // TODO - remove the need for this
+                httpify(deferred);
+
+                if(pendingGETRequests[resourceName]){
+                    // theres already a pending request to
+                    // this endpoint, so fail!
+                    deferred.reject(`a request to ${resourceName} is pending`);
```

Comment:
Maybe I’m missing something, but it looks like it doesn’t exactly backoff (e.g. poll slower). Instead it just doesn’t poll at all, correct?   I guess that's OK - just wanted to confirm the strategy.